src/Engine/UnitActionLifecycles/LaunchLifecycle.java

```diff
@@ -0,0 +1,212 @@
+import Terrain.MapMaster;
+import UI.MapView;
+import UI.Art.Animation.GameAnimation;
+import Units.Unit;
+
+public abstract class LaunchLifecycle
+{
+  public static class LaunchFactory extends UnitActionFactory
+  {
+    private static final long serialVersionUID = 1L;
+
+    @Override
+    public GameActionSet getPossibleActions(GameMap map, Path movePath, Unit actor, boolean ignoreResident)
+    {
+      XYCoord moveLocation = movePath.getEndCoord();
+      if( moveLocation.equals(actor.x, actor.y) )
+      {
+        if( actor.heldUnits.size() > 0 )
+        {
+          ArrayList<GameAction> launchActions = new ArrayList<GameAction>();
+
+          for( Unit cargo : actor.heldUnits )
+          {
+            if( cargo.isTurnOver )
+              continue; // No acting on the same turn as you load
+
+            // For the benefit of our planning functions, consider the cargo to be on the launcher's space...
+            cargo.x = actor.x;
+            cargo.y = actor.y;
+
+            ArrayList<XYCoord> destinations = Utils.findPossibleDestinations(cargo, map);
+            // Acting in place after a launch is a cool concept, but a little weird in execution
+            // Also, allowing action in place would allow *launching*, and recursive launching isn't something I wanna bite off
+            destinations.remove(moveLocation);
+
+            // Build a launch action for each possible action the cargo can do after launch
+            for( XYCoord coord : destinations )
+            {
+              Path cargoMovePath = Utils.findShortestPath(cargo, coord, map);
+              ArrayList<GameActionSet> cargoActions = cargo.getPossibleActions(map, cargoMovePath, ignoreResident);
+
+              for( GameActionSet actionSet : cargoActions )
+                for( GameAction action : actionSet.getGameActions() )
+                  launchActions.add(new LaunchAction(actor, cargo, action));
+            }
+
+            // ...and reset it when we're done calculating
+            cargo.x = -1;
+            cargo.y = -1;
+          }
+
+          if( !launchActions.isEmpty() )
+          {
+            return new GameActionSet(launchActions);
+          }
+        }
+      }
+      return null;
+    }
+
+    @Override
+    public String name()
+    {
+      return "LAUNCH";
+    }
+
+    /**
+     * From Serializable interface
+     * @return The statically-defined object to use for this action type.
+     */
+    private Object readResolve()
+    {
+      return LAUNCH;
+    }
+  } // ~Factory
+
+  public static class LaunchAction implements GameAction
+  {
+    private Unit launcher, launchee;
+    private GameAction cargoAction;
+
+    public LaunchAction(Unit actor, Unit passenger, GameAction passengerAction)
+    {
+      launcher = actor;
+      launchee = passenger;
+      cargoAction = passengerAction;
+    }
+
+    @Override
+    public GameEventQueue getEvents(MapMaster gameMap)
+    {
+      // UNLOAD actions consist of
```

Comment:
This appears to be pasta'd from UNLOAD.

Reply:
Bah. Coulda sworn I caught that one.
Probably ended up ctrl-z'ing it and didn't notice.